main/zoe/guide/contracts/multipoolAutoswap.md

```diff
@@ -0,0 +1,169 @@
+appropriate pools don't exist, she'll get an error (`brands were not recognized`).
+
+```js
+const saraProposal = harden({
+  want: { out: simoleans(275) },
+  give: { in: moola(220) },
+});
+
+const swapInvitation = await E(publicFacet).makeSwapOutInvitation();
+const moolaPayment = harden({ In: sMoolaPurse.withdraw(220) });
+
+const saraSeat = await E(zoe).offer(swapInvitation, saraProposal, moolaPayment);
+const saraResult = await saraSeat.getOfferResult();
+```
+
+If the result is `Swap successfully completed.`, she got the simoleans for 220 or
+less (she'll want to deposit any refund). Otherwise the market price moved against
+her, and she'll have to check the price again and make another offer.
+
+```js
+const moolaRefund = await E(saraSeat).getPayout('In');
+const simoleanProceeds = await E(saraSeat).getPayout('Out');
+
+const moolaRefundAmount = E(sMoolaPurse).deposit(moolaRefund);
+E(simoleanPurse).deposit(simoleanProceeds);
+```
+
+###  Creating a new Pool
+
+When the contract is first instantiated, there won't be any pools ready for
+ trading. `addPool()` adds a new currency, which can then be funded. When a pool is
+first funded, there's no other basis on which to decide how much liquidity to create,
+so the liquidity amount equals the amount of the central token in the offer.
+
+```js
+const moolaLiquidityIssuer = await E(publicFacet).addPool(moolaIssuer, 'Moola');
+const moolaLiquidityAmountMath = await makeLocalAmountMath(moolaLiquidityIssuer);
+```
+
+Alice decides that the current rate in the external market is 2 Moola for each
+Buck, so she deposits twice as many Moola as Bucks.
+
+```js
+const aliceProposal = harden({
+  want: { Liquidity: moolaLiquidity(50) },
+  give: {
+    Secondary: moolaAmountMath.make(100),
+    Central: bucksAmountMath.make(50),
+  },
+});
+const alicePayments = {
+  Secondary: aliceMoolaPayment,
+  Central: aliceBucksPayment,
+};
+
+const aliceAddLiquidityInvitation = E(publicFacet).makeAddLiquidityInvitation();
+const addLiquiditySeat = await zoe.offer(
+  aliceAddLiquidityInvitation,
+  aliceProposal,
+  alicePayments,
+);
+```
+
+### Adding Liquidity to an Existing Pool
+
+When adding or removing liquidity to pools that have already been established, the
+amounts deposited must be in proportion to the current balances in the pool. The
+calculation is based on the amount of the `Central` asset. The `Secondary` assets
+must be added in proportion.  If less `Secondary` is provided than required, the
+offer is exited with no trade. If more of the secondary is provided than is required,
+the excess is returned.
+
+Bob calls `getPoolAllocation()` to find the relative levels. Let's say the answer is
+that the current ratio is 1234 moola to 1718 bucks.
+
+```js
+const moolaPoolAlloc = E(publicFacet).getPoolAllocation(moolaBrand);
+const bucksValue = moolaPoolAlloc.Central.value;
+const moolaValue = moolaPoolAlloc.secondary.value;
+```
```

Comment:
This is out of the scope for the docs, but I wonder if we need a better mechanism for figuring out how much secondary is necessary. Can we use one of the price methods?